script/tool/lib/src/pubspec_check_command.dart

```diff
@@ -352,6 +352,15 @@ class PubspecCheckCommand extends PackageLoopingCommand {
             'a topic. Add "$topicName" to the "topics" section.';
       }
     }
+    for (final String topic in topics) {
+      final RegExp expectedTopicFormat =
+          RegExp(r'^[a-z](?:[a-z0-9]*[a-z0-9])?(-[a-z0-9]+)*$');
```

Comment:
Couldn't we make the `-` optional rather than need the middle group? So `r'[a-z](?:-?[a-z0-9]+)*$'`?

Reply:
Oh you're right lol. I may have over thought this a little bit. I think your example with a '^' appended in the beginning is functionally equivalent haha.